src/main/java/frc/robot/commands/auto/PathPlannerCommand.java

```diff
@@ -54,16 +60,29 @@ public PathPlannerCommand(List<PathPlannerTrajectory> pathGroup, int pathIndex,
         drive.resetOdometry(Conversions.absolutePoseToPathPlannerPose(path.getInitialHolonomicPose(), DriverStation.getAlliance()));
       }
     }), 
-    new PPSwerveControllerCommand(
-      pathGroup.get(pathIndex), 
-      () -> Conversions.absolutePoseToPathPlannerPose(
-      drive.getPose(), DriverStation.getAlliance()), // Pose supplier
-      drive.getPathplannerXController(), // X controller can't normal PID as pathplanner has Feed Forward 
-      drive.getPathplannerYController(), // Y controller can't normal PID as pathplanner has Feed Forward 
-      drive.getPathplannerRotationController(), // Rotation controller can't normal PID as pathplanner has Feed Forward 
-      (chassisSpeeds) -> { drive.setChassisSpeeds(chassisSpeeds, false); }, // chassis Speeds consumer
-      true,  // use Alliance color
-      drive // Requires this drive subsystem
+    (m_perpetual ? 
+      new PPSwerveControllerCommandPerpetual(
+        pathGroup.get(pathIndex), 
+        () -> Conversions.absolutePoseToPathPlannerPose(
+        drive.getPose(), DriverStation.getAlliance()), // Pose supplier
+        drive.getPathplannerXController(), // X controller can't normal PID as pathplanner has Feed Forward 
+        drive.getPathplannerYController(), // Y controller can't normal PID as pathplanner has Feed Forward 
+        drive.getPathplannerRotationController(), // Rotation controller can't normal PID as pathplanner has Feed Forward 
+        (chassisSpeeds) -> { drive.setChassisSpeeds(chassisSpeeds, false); }, // chassis Speeds consumer
+        true,  // use Alliance color
+        drive // Requires this drive subsystem
+      ) :
+      new PPSwerveControllerCommand(
+        pathGroup.get(pathIndex), 
+        () -> Conversions.absolutePoseToPathPlannerPose(
+        drive.getPose(), DriverStation.getAlliance()), // Pose supplier
+        drive.getPathplannerXController(), // X controller can't normal PID as pathplanner has Feed Forward 
+        drive.getPathplannerYController(), // Y controller can't normal PID as pathplanner has Feed Forward 
+        drive.getPathplannerRotationController(), // Rotation controller can't normal PID as pathplanner has Feed Forward 
+        (chassisSpeeds) -> { drive.setChassisSpeeds(chassisSpeeds, false); }, // chassis Speeds consumer
+        true,  // use Alliance color
+        drive // Requires this drive subsystem
+      )
```

Comment:
I feel like this could be cleaner. Maybe make a static function in PPSwerveControllerCommandPerpetual that returns either a the perpetual version or the normal version based on a bool (along with taking all the other params). that prevents errors in case someone modifies something in one of the parameters but forgets to in the other parameter list.
so it would be like
PathPlannerCommand.java:
```
...
addCommands(
 new InstantCommand(...), ...
 PPSwerveControllerCommandPerpetual.create(path, pose, .... , isPerpetual)
);
```
PPSwerveControllerCommandPerpetual.java
```
public static PPSwerveControllerCommand create(Path path, pose pose, ... boolean isPerpetual) {
   if (isPerpetual) {
        // the perpetual version
        return new PPSwerveControllerCommandPerpetual(path, pose, ...);
   } else {
        // the normal version
       return new PPSwerveControllerCommand(path, pose, ...);
  }
}
```

Reply:
I think I did what you were seggusting. But I am still not very sure exactly what you ment.

Reply:
That's not what I was suggesting. You need a static function in PPSwerveControllerCommandPerpetual like the one I wrote. I can talk to you about it in class

Reply:
Not exactly. I wanted all the parameters that might change to be inputted in one place. I did that